ignite/metrics/nlp/bleu.py

```diff
@@ -189,16 +284,35 @@ def _corpus_bleu(
     def reset(self) -> None:
```

Comment:
Let's split the variables based on the `average` value.

Reply:
Sorry I didn't understand what you mean by this.

Reply:
I mean `_sum_of_bleu` and `_num_sentences` is related to macro, while the others to micro. Don’t instanciate useless variables if possible.